src/Discovery/src/EurekaBase/EurekaDiscoveryClientExtension.cs

```diff
@@ -0,0 +1,109 @@
+                : GetNamedDiscoveryServiceInfo(config, ServiceInfoName);
+
+            ConfigureEurekaServices(services, config, info, netOptions);
+            AddEurekaServices(services, lifecyle);
+        }
+
+        private static void ConfigureEurekaServices(IServiceCollection services, IConfiguration config, IServiceInfo info, InetOptions netOptions)
+        {
+            var einfo = info as EurekaServiceInfo;
+            var clientSection = config.GetSection(EurekaClientOptions.EUREKA_CLIENT_CONFIGURATION_PREFIX);
+            services.Configure<EurekaClientOptions>(clientSection);
+            services.PostConfigure<EurekaClientOptions>((options) =>
+            {
+                EurekaPostConfigurer.UpdateConfiguration(config, einfo, options);
+            });
+
+            var instSection = config.GetSection(EurekaInstanceOptions.EUREKA_INSTANCE_CONFIGURATION_PREFIX);
+            services.Configure<EurekaInstanceOptions>(instSection);
+            services.PostConfigure<EurekaInstanceOptions>((options) =>
+            {
+                IApplicationInstanceInfo appInfo = null;
+                if (einfo?.ApplicationInfo == null)
+                {
+                    appInfo = services.GetApplicationInstanceInfo();
+                }
+
+                options.NetUtils = new InetUtils(netOptions);
+                options.ApplyNetUtils();
+                EurekaPostConfigurer.UpdateConfiguration(config, einfo, options, einfo?.ApplicationInfo ?? appInfo);
+            });
+        }
+
+        private static void AddEurekaServices(IServiceCollection services, IDiscoveryLifecycle lifecycle)
+        {
+            services.AddSingleton<EurekaApplicationInfoManager>();
+            services.AddSingleton<EurekaDiscoveryManager>();
+
+            services.AddSingleton<EurekaDiscoveryClient>();
+            if (lifecycle == null)
+            {
+                services.AddSingleton<IDiscoveryLifecycle, ApplicationLifecycle>();
+            }
+            else
+            {
+                services.AddSingleton(lifecycle);
+            }
+
+            services.AddSingleton<IDiscoveryClient>((p) =>
+            {
+                var eurekaService = p.GetService<EurekaDiscoveryClient>();
+
+                // Wire in health checker if present
+                if (eurekaService != null)
+                {
+                    eurekaService.HealthCheckHandler = p.GetService<IHealthCheckHandler>();
+                }
+
+                return eurekaService;
+            });
+
+            services.AddSingleton<IHealthContributor, EurekaServerHealthContributor>();
+
+            var serviceProvider = services.BuildServiceProvider();
+            var certOptions = serviceProvider.GetService<IOptions<CertificateOptions>>();
+            var existingHandler = serviceProvider.GetService<IHttpClientHandlerProvider>();
+            if (certOptions != null && existingHandler is null)
```

Comment:
is this mix of null/not null a common csharp practice?

Alternatives that _seem_ more consistent:
```
(certOptions != null && existingHandler == null)
(!(certOptions is null) && existingHandler is null)
```

Reply:
```suggestion
            if (certOptions is object && existingHandler is null)
```
StackOverflow has this as a suggestion... thoughts?

Reply:
looks good